File: MaWPhotos/src/main/java/us/mikeandwan/photos/ui/categories/ListCategoryRecyclerAdapter.java
package us.mikeandwan.photos.ui.categories;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.squareup.picasso.Picasso;

import butterknife.BindView;
import butterknife.ButterKnife;
import us.mikeandwan.photos.R;
import us.mikeandwan.photos.models.Category;
import us.mikeandwan.photos.services.DataServices;


public class ListCategoryRecyclerAdapter extends CategoryRecyclerAdapter<ListCategoryRecyclerAdapter.ViewHolder> {
    public ListCategoryRecyclerAdapter(ICategoryListActivity activity,
                                       DataServices dataServices) {
        super(activity, dataServices);
    }


    @Override
    public ListCategoryRecyclerAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View categoryView = LayoutInflater.from(parent.getContext()).inflate(R.layout.category_list_item, parent, false);

        return new ViewHolder(categoryView);
    }


    @Override
    protected String downloadCategoryTeaser(Category category) {
        return _dataServices.downloadCategoryTeaser(category);
    }


    protected void displayCategory(Category category, String path, ListCategoryRecyclerAdapter.ViewHolder viewHolder) {
        viewHolder._nameTextView.setText(category.getName());

        Picasso
                .with(_context)
                .load(path)
                .resizeDimen(R.dimen.category_list_thumbnail_size, R.dimen.category_list_thumbnail_size)
                .centerCrop()
                .into(viewHolder._thumbnailImageView);
    }


    static class ViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.thumbnailImageView) ImageView _thumbnailImageView;
        @BindView(R.id.categoryNameTextView) TextView _nameTextView;


        ViewHolder(View itemView) {
            super(itemView);

            ButterKnife.bind(this, itemView);
        }
    }
}

File: MaWPhotos/src/main/java/us/mikeandwan/photos/ui/categories/ListCategoryRecyclerAdapter.kt
package us.mikeandwan.photos.ui.categories

import us.mikeandwan.photos.services.DataServices
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import us.mikeandwan.photos.R
import com.squareup.picasso.Picasso
import androidx.recyclerview.widget.RecyclerView
import android.widget.TextView
import us.mikeandwan.photos.databinding.CategoryListItemBinding
import us.mikeandwan.photos.models.Category

class ListCategoryRecyclerAdapter(
    activity: ICategoryListActivity?,
    dataServices: DataServices?
) : CategoryRecyclerAdapter<ListCategoryRecyclerAdapter.ViewHolder?>(
    activity!!, dataServices!!
) {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = CategoryListItemBinding.inflate(LayoutInflater.from(parent.context), parent, false)

        return ViewHolder(binding)
    }

    override fun downloadCategoryTeaser(category: Category?): String? {
        return _dataServices.downloadCategoryTeaser(category)
    }

    override fun displayCategory(
        category: Category?,
        path: String?,
        viewHolder: ViewHolder?
    ) {
        viewHolder!!._nameTextView!!.text = category!!.name
        Picasso
            .get()
            .load(path)
            .resizeDimen(R.dimen.category_list_thumbnail_size, R.dimen.category_list_thumbnail_size)
            .centerCrop()
            .into(viewHolder._thumbnailImageView)
    }

    class ViewHolder(private val binding: CategoryListItemBinding) : RecyclerView.ViewHolder(binding.root) {
        var _thumbnailImageView = binding.thumbnailImageView
        var _nameTextView = binding.categoryNameTextView
    }
}